File: Modules/ServerTypes/TaskFlow/TaskColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TaskPriority } from "./TaskPriority";
import { TaskRow } from "./TaskRow";

export interface TaskColumns {
    Title: Column<TaskRow>;
    ProjectName: Column<TaskRow>;
    Description: Column<TaskRow>;
    TaskStateName: Column<TaskRow>;
    Priority: Column<TaskRow>;
    TagNames: Column<TaskRow>;
    DueDate: Column<TaskRow>;
    Duration: Column<TaskRow>;
    CreatedAt: Column<TaskRow>;
    UpdatedAt: Column<TaskRow>;
}

export class TaskColumns extends ColumnsBase<TaskRow> {
    static readonly columnsKey = 'TaskFlow.Task';
    static readonly Fields = fieldsProxy<TaskColumns>();
}

[TaskPriority]; // referenced types

File: Modules/TaskFlow/Task/TaskColumns.cs
namespace TaskFlowSerenity.TaskFlow.Columns;

[ColumnsScript("TaskFlow.Task")]
[BasedOnRow(typeof(TaskRow), CheckNames = true)]
public class TaskColumns
{
    [EditLink]
    public string Title { get; set; }
    public string ProjectName { get; set; }
    public string Description { get; set; }
    public string TaskStateName { get; set; }

    [QuickFilter]
    public TaskPriority Priority { get; set; }
    [DisplayName("Tags")]
    public string TagNames { get; set; }
    public DateTime DueDate { get; set; }

    [DisplayName("Duration (Hours)")]
    public int Duration { get; set; }

    [SortOrder(1, descending: true)]
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}